pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -507,8 +501,13 @@ func (r *ReconcileNetworkAddonsConfig) trackDeployedObjects(objs []*unstructured
 
 	r.podReconciler.SetResources(allResources)
 
-	// Trigger status manager to notice the change
-	r.statusManager.SetFromPods()
+	// From now on, r.podReconciler takes over NetworkAddonsConfig handling, it will track deployed
+	// objects if needed and set NetworkAddonsConfig.Status accordingly. However, if no resources were
+	// deployed, there is nothing that would trigger initial reconciliation. Therefore, let's
+	// perform the first check manually.
+	if len(allResources) == 0 {
```

Comment:
shouldn't it be != 0 ? (according comment "However, if no resources were deployed, there is nothing that would trigger initial reconciliation")

Reply:
ah.. I don't think so..
`if no resources were deployed..` i.e. len(allResources) == 0, `..there is nothing that would trigger initial reconciliation.`
`Therefore, let's perform the first check manually.`, then run r.statusManager.SetFromPods() manually, and not from the podController.